pkg/kv/kvserver/batcheval/cmd_refresh_range.go

```diff
@@ -89,7 +89,7 @@ func refreshRange(
 			break
 		}
 
-		key := iter.Key()
+		key := iter.UnsafeKey().Clone()
```

Comment:
What is the reason to chose calling Clone() vs using Key() which is meant to return a durable copy of underlying value? Should we just make iterator use Clone() under the hood when returning Key() or we want to deprecate that completely?

Reply:
`MVCCIncrementalIterator` only implements `SimpleMVCCIterator`, and `Key()` is not a method on `SimpleMVCCIterator` -- it's on `MVCCIterator`.

I don't have a strong opinion on `MVCCIterator.Key()` -- I think I'd prefer to get rid of it just to reduce the API surface, since it's trivial to clone when needed (every single MVCC iterator needs to implement this cloning). But for the purposes of this PR, the idea was just to make `MVCCIncrementalIterator` conform to `SimpleMVCCIterator`, which is the only interface it implements.

Reply:
I like the idea making it unsafe and letting caller do cloning. We may put that in the comment specifically that you should use `.Clone()` on result if you want to save it or do `safe = append(safe[0:], it.UnsafeKey())` depending on how key is reused later.

Reply:
Yeah. We're going to be overhauling the `MVCCIterator` interface over in #82589, so I added a comment there to consider getting rid of them.